lib/logstash/outputs/elasticsearch/http_client.rb

```diff
@@ -348,6 +348,37 @@ def template_put(name, template)
       @pool.put(path, nil, LogStash::Json.dump(template))
     end
 
+    def write_alias_exists?(name)
```

Comment:
Consider refactoring this to `es_resource_exists?` and calling it from `template _exists?` as well as `maybe_create_ilm_policy` and `maybe_create_write_alias ` so that these last two read
`plugin.client.es_resource_exists?(ilm_write_alias)` as it is quite clear what you are checking the existence of.

Also, do you need interpolation and the creation of a new string object in `@pool.head("#{name}")`?